### PR TITLE
Extends dedicated_southbound_right_turn_lane.

### LINKED ISSUE #43: Map request: Dedicated turn lane going south with extended approach
This is a second iteration of #27. We'd like a version of this map with extended approach and exit stretches of road. See image below.

dedicated_turn_lane_going_south_extended_approach

### PR DESCRIPTION
**Pull request by francocipollone**
# 🎉 New feature

Closes #43 

## Summary
 - Extends approaching roads
 - Adds resource as required.
 - Renames map from `dedicated_turn_lane_going_south` to `dedicated_southbound_right_turn_lane`
   - `resources/dedicated_southbound_right_turn_lane/dedicated_southbound_right_turn_lane.xodr` is the requested one with 300m of extension
      - image

   - `resources/dedicated_southbound_right_turn_lane/dedicated_southbound_right_turn_lane_100m.xodr` is the old version

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by liangfok** [comment]
@francocipollone can you resolve the merge conflict? Looks like it's conflicting with ce7ffcb9096633038ef67ae0ed8a500d4c742954.

**Comment by francocipollone** [comment]
> @francocipollone can you resolve the merge conflict? Looks like it's conflicting with [ce7ffcb](https://github.com/maliput/maliput_xodr/commit/ce7ffcb9096633038ef67ae0ed8a500d4c742954).

Conflict is solved

**Review by liangfok — DISMISSED** [review]
LGTM

**Comment by liangfok** [comment]
Sorry for the last-minute request, but we encountered a downstream issue and suspect adding the transition from the east-bound lane to the north-bound lane as indicated in the image below might help.
 
dedicated_right_turn_lane_map
 
If it makes sense, feel free to merged this PR first. We can always issue another PR that includes another version of this map with the indicated transition modeled.

File: resources/dedicated_southbound_right_turn_lane/dedicated_southbound_right_turn_lane.xodr [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+    <junction id="10" name="Junction 10">
+        <connection id="0" incomingRoad="1" connectingRoad="2" contactPoint="start">
+            <laneLink from="1" to="1"/>
+        </connection>
+        <connection id="1" incomingRoad="4" connectingRoad="5" contactPoint="end">
+            <laneLink from="-1" to="1"/>
+        </connection>
+        <connection id="2" incomingRoad="4" connectingRoad="6" contactPoint="start">
+            <laneLink from="1" to="1"/>
+        </connection>
```
[the remark this answers is not in the document]
Comment (francocipollone) [reply resources/dedicated_southbound_right_turn_lane/dedicated_southbound_right_turn_lane.xodr:431]: Correct, that transition is already added in the map

image

**Comment by liangfok** [comment]
LGTM

**Pull request merged**